search/intermediate_result.cpp

```diff
@@ -282,19 +283,42 @@ void FillDetails(FeatureType & ft, Result::Details & details)
 
   feature::TypesHolder const typesHolder(ft);
 
-  details.m_isHotel = ftypes::IsHotelChecker::Instance()(typesHolder);
-  if (details.m_isHotel && strings::to_uint(ft.GetMetadata(feature::Metadata::FMD_STARS), details.m_stars))
-    details.m_stars = std::min(details.m_stars, osm::MapObject::kMaxStarsCount);
-  else
-    details.m_stars = 0;
+  uint8_t starsCount = 0;
+  bool isHotel = ftypes::IsHotelChecker::Instance()(typesHolder);
+  if (isHotel && strings::to_uint(ft.GetMetadata(feature::Metadata::FMD_STARS), starsCount))
+    starsCount = std::min(starsCount, osm::MapObject::kMaxStarsCount);
 
   auto const cuisines = feature::GetLocalizedCuisines(typesHolder);
-  details.m_cuisine = strings::JoinStrings(cuisines, osm::MapObject::kFieldsSeparator);
+  std::string cuisine = strings::JoinStrings(cuisines, osm::MapObject::kFieldsSeparator);
 
   auto const roadShields = ftypes::GetRoadShieldsNames(ft);
-  details.m_roadShields = strings::JoinStrings(roadShields, osm::MapObject::kFieldsSeparator);
+  std::string roadShield = strings::JoinStrings(roadShields, osm::MapObject::kFieldsSeparator);
 
-  details.m_fee = feature::GetLocalizedFeeType(typesHolder);
+  std::string fee = feature::GetLocalizedFeeType(typesHolder);
+    
+  std::string description;
+  
+  if (isHotel && starsCount > 0)
+  {
+    std::string stars;
+    for (int i = 0; i < starsCount; ++i)
+      stars.append(osm::MapObject::kStarSymbol);
+    description = stars;
+  }
+  else if (airportIata.length() > 0)
+    description = airportIata;
+  else if (roadShield.length() > 0)
+    description = roadShield;
+  else if (brand.length() > 0 && cuisine.length() > 0)
+    description = brand + osm::MapObject::kFieldsSeparator + cuisine;
+  else if (brand.length() > 0)
+    description = platform::GetLocalizedBrandName(brand);
```

Comment:
0. `!str.empty()` instead of checking the length is shorter.
1. What is the difference for brands here? Why cuisine can't be added later?
2. What about hotels? Should brand/cuisine be displayed there too? If yes, then the first if won't allow that.
3. If brand value is equal to the title, then let's not display it in the description/subtitle.

Reply:
Changed to !str.empty() and brands/cuisine are appended just the same as the others